apps/server/src/utilities/network.service.ts

```diff
@@ -0,0 +1,190 @@
+import { AddressInfo, createConnection } from 'net';
+import { platform } from 'os';
+import { execSync } from 'child_process';
+import { Injectable, Logger, OnApplicationShutdown } from '@nestjs/common';
+import * as createMDNSServer from 'multicast-dns';
+import { MulticastDNS, QueryPacket } from 'multicast-dns';
+import { StringAnswer } from 'dns-packet';
+import { RemoteInfo } from 'dgram';
+
+@Injectable()
+export class NetworkService implements OnApplicationShutdown {
+    private readonly logger = new Logger(NetworkService.name);
+
+    private mDNSServer: MulticastDNS | undefined;
+
+    constructor() {
+        this.createMDNSServer();
+    }
+
+    async createMDNSServer() {
```

Comment:
Can this renamed, it can cause confusion as it's the same name as the imported `multicast-dns` module